lsp-ui-doc.el

```diff
@@ -776,11 +761,13 @@ before, or if the new window is the minibuffer."
         (cl-callf copy-tree frameset-filter-alist)
         (push '(lsp-ui-doc-frame . :never) frameset-filter-alist)))
     (add-hook 'post-command-hook 'lsp-ui-doc--make-request nil t)
-    (add-hook 'delete-frame-functions 'lsp-ui-doc--on-delete nil t))
+    (add-hook 'delete-frame-functions 'lsp-ui-doc--on-delete nil t)
+    (remove-hook 'focus-in-hook 'posframe--redirect-posframe-focus t))
```

Comment:
The doc frame focus works really spotty with this PR, especially when using together with `flycheck-mode`.
I have to press `lsp-ui-doc-focus-frame` key bind several times before I can sucessfully focus into doc frame. Moreover, each time it failed, the doc frame will go away and I need to trigger it showing again.

The reason seems to be `focus-in-hook` become local variable and thus `posframe--redirect-posframe-focus` is only removed from local value, and still being triggered in global value (my guess).

I think instead of removing the function from hook, it might be better to just advice reroute it when `lsp-ui-doc-frame-mode` is on. Some thing like this
```suggestion
    (advice-add 'posframe--redirect-posframe-focus
                :before-until (lambda (&rest _)
                                lsp-ui-doc-frame-mode)
                '((name . lsp--posframe-dont-redirect-on-doc-frame)))
```